pkg/kn/flags/bool.go

```diff
@@ -48,10 +48,11 @@ func AddBothBoolFlags(f *pflag.FlagSet, p *bool, name, short string, value bool,
 }
 
 // ReconcileBoolFlags sets the value of the all the "--foo" flags based on
-// "--no-foo" if provided, and returns an error if both were provided.
+// "--no-foo" if provided, and returns an error if both were provided or an
+// explicit value of false was provided to either (as that's confusing).
 func ReconcileBoolFlags(f *pflag.FlagSet) error {
 	var err error
-	f.Visit(func(flag *pflag.Flag) {
+	f.VisitAll(func(flag *pflag.Flag) {
 		// Return early from our comprehension
 		if err != nil {
 			return
```

Comment:
Style nit... It's interesting that you are simply using `return` here without explicitly stating the `error` being returned and have not defined the function with a named return --- something I have never been a fan of.

So to be clear, I'd expect one of two things: 
a) `return err`

or 

b) `return` if the function was defined as `func ReconcileBoolFlags(f *pflag.FlagSet) (err error)`

Maybe I am missing something... obviously am since this compiles. Anyhow.

Reply:
Oh I see this return is for the `VisitAll` anonymous function. Oh life makes sense again :) Ignore above.